architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md

```diff
@@ -0,0 +1,230 @@
+
+Now on creation, the schema should be validated for correctness and completeness. Deb should be provided with error messages if the schema is incorrect or incomplete.
+
+E.g.:
+
+    ```bash
+    rad resource-type create -f plaid.yaml
+    Error: The schema definition is incorrect. The required property 'port' is missing a type in the schema definition
+    ```
+
+### Scenario 2: Deb updates the schema for the internal messaging service Plaid
+<!-- One or two sentence summary -->
+Updates to schema definitions can be classified into two categories: non-breaking changes and breaking changes. 
+
+| Change type | Description | Example |
+|-------------|-------------|---------|
+| Non-breaking changes | Changes that can be made to the schema definition without impacting existing applications that use the resource type. | Eg: Adding an optional property to the schema definition |
+| Breaking changes | Changes that are not backward/forward compatible. These changes are disruptive and often requires the users to update their applications to use the new schema definition. | Eg: Adding a required property to the schema definition |
+
+There are two key sub-scenarios to consider when Deb updates the schema for the internal messaging service Plaid
+
+#### Sub-scenario 1: Deb adds a optional property and updates the resource type Plaid.
+Deb wants to add an optional property `logAnalytics` to have log analytics enabled for the resource-type. This is considered to be a non-breaking change as it is an optional property with default value. Users should be able to use the new property without any impact across versions on their existing applications.
+
+```yaml
+namespace: 'Mycompany.Messaging'
+resourceTypes:
+    'Mycompany.Messaging/plaidResource':
+        name: 'Name of the resource'
+        version: '2024-11-01'
+            properties: 
+                required: ['queueName','host', 'port', 'connectionString']
+                queueName:
+                    type: 'string'
+                    description: 'Name of the queue'       
+                port:
+                    type: 'string'
+                    description: 'Port'
+                connectionString:
+                    type: 'string'
+                    description: 'Connection string to the messaging service'
+                logAnalytics:
+                    type: 'string'
+                    description: 'Usage type of the messaging service'  
+                    default: 'off'        
+```
+
+Deb creates the updated resource type schema via the CLI
+
+```bash
+rad resource-type create -f plaid.yaml
+Resource type `plaid` created successfully for version `2024-11-01`
+```
+
+Question 
+
+1. We leave it up to the user to decide whether to use the new optional property in their application definitions and Recipes. Should we provide any guidance?
```

Comment:
I don't understand the question. Optional is optional.

Reply:
Sorry the question is incomplete. I was thinking about guidance/feature from GitOps perspective. Is that something we should provide or think about ? How Gitops tools understand changes to the Radius resource-type schemas ?